addons/mail/models/mail_thread.py

```diff
@@ -389,6 +390,25 @@ def get_empty_list_help(self, help_message):
 
         return super().get_empty_list_help(help_message)
 
+    def _flush_search(self, domain, fields=None, order=None, seen=None):
+        domain_message_partner_ids, other_domain = partition(
+            lambda leaf: len(leaf) == 3 and leaf[0] == 'message_partner_ids',
+            domain,
+        )
```

Comment:
As `partition` filters out `message_partner_ids`-related leafs, the list `other_domain` might not be a valid domain.  I don't know whether it is relevant here, but it may be a source of bugs...
Also, what about leaves like `('message_partner_ids.name', 'ilike', ...)` for instance?  As far as I can tell, those will be returned into `other_domain` and subject to regular "group" restrictions.  If it is intended, it deserves to be documented.

Reply:
You are absolutely right. `BaseModel.flush_domain` doesn't care about having a valid domain anyway, but it's not nice to break the validity of the domain as well. I am going to change the behavior: replace the leaf with 'message_partner_ids' with a dummy one in order to keep a consistent domain.

In fact, `('message_partner_ids.name', 'ilike', ...)` needs to be checked.